### ISSUE TITLE
[BUG] Fully shutdown then power on harvester node machine can't get provisioned RKE2 cluster back to work

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1561]
**Describe the bug**

Fully power off harvester node machine, wait for a while then power on. 
The RKE2 cluster provisioning in rancher can't turn back to running 
Keep in `Cluster health check failed: Failed to communicate with API server during namespace check`

![image.png](https://images.zenhubusercontent.com/61519853321ea20d65443929/40eefc37-c5ac-46a6-81e8-9e7fe8b7bd85)

**To Reproduce**
1. Installed a 3 nodes harvester cluster 

2. Install rancher 2.6.2 from docker 
3. Import harvester to rancher in virtualization management 
4. Enable node driver and create cloud credential
5. Provision a RKE2 cluster in rancher 
6. Confirm RKE2 cluster is fully operated, can explore it 
![image.png](https://images.zenhubusercontent.com/61519853321ea20d65443929/b29e812c-1430-4447-9e34-a52a92895140)
7. Shutdown all 3 nodes server machine
![image.png](https://images.zenhubusercontent.com/61519853321ea20d65443929/c351d09f-4745-4b87-b9a4-30f91cb4ab1f)
8. Wait for 10 minutes
9. Power on all harvester nodes server machines
10. Confirm harvester is fully operated
11. Confirm RKE2 vm is back to running
![image.png](https://images.zenhubusercontent.com/61519853321ea20d65443929/a97b7464-747b-4fee-8368-651270705106)
12. Check the RKE2 cluster status in rancher  

**Expected behavior**

The RKE2 cluster in rancher should turn back to `Running` with no error after harvester server node machine is fully power off and power on. 

**Support bundle**

**Environment:**
 - Harvester ISO version: master-45e25939-head
-  Rancher version: 2.6.2
 - Underlying Infrastructure: 3 nodes harvester on local kvm/qemu machine

**Comment by futuretea** [comment]
My scenario is a little different, after power-off/on repeatly

vm will become running, 
![image](https://user-images.githubusercontent.com/15064560/143063974-8c2d0026-428c-48d1-a0a2-80739161882a.png)
but in the guest cluster, many pods stuck in state `Unknown`
```bash
NAMESPACE             NAME                                                   READY   STATUS      RESTARTS   AGE
calico-system         calico-kube-controllers-748bcb7bb-2xzgw                0/1     Unknown     0          84m
calico-system         calico-node-kpd9k                                      1/1     Running     2          84m
calico-system         calico-typha-5d79b854d8-ngksw                          1/1     Running     2          84m
cattle-fleet-system   fleet-agent-8596fc94d5-xnzxr                           0/1     Unknown     0          76m
cattle-system         cattle-cluster-agent-77f65fd955-jqqqj                  0/1     Unknown     1          63m
cattle-system         system-upgrade-controller-7c878c4798-qtxc6             0/1     Unknown     0          76m
kube-system           etcd-demo-pool1-ec74d5f5-lqkkw                         1/1     Running     2          87m
kube-system           harvester-cloud-provider-5bc9dcbb77-4qrwv              0/1     Unknown     0          85m
kube-system           harvester-cloud-provider-5bc9dcbb77-b4tlx              0/1     Unknown     0          85m
kube-system           harvester-csi-driver-controllers-fcd95788c-9x48h       0/3     Unknown     0          85m
kube-system           harvester-csi-driver-controllers-fcd95788c-bzq9z       0/3     Unknown     0          85m
kube-system           harvester-csi-driver-controllers-fcd95788c-w6swx       0/3     Unknown     0          85m
kube-system           harvester-csi-driver-tgdq9                             0/2     Unknown     0          82m
kube-system           helm-install-harvester-cloud-provider-5dvfl            0/1     Completed   0          87m
kube-system           helm-install-harvester-csi-driver-w4vqv                0/1     Completed   0          87m
kube-system           helm-install-rke2-calico-crd-75zgz                     0/1     Completed   0          87m
kube-system           helm-install-rke2-calico-tbqkl                         0/1     Completed   2          87m
kube-system           helm-install-rke2-coredns-fxd5s                        0/1     Completed   0          87m
kube-system           helm-install-rke2-ingress-nginx-d55sw                  0/1     Completed   0          87m
kube-system           helm-install-rke2-metrics-server-8m7xp                 0/1     Completed   0          87m
kube-system           kube-apiserver-demo-pool1-ec74d5f5-lqkkw               1/1     Running     2          87m
kube-system           kube-controller-manager-demo-pool1-ec74d5f5-lqkkw      1/1     Running     2          87m
kube-system           kube-proxy-demo-pool1-ec74d5f5-lqkkw                   1/1     Running     2          87m
kube-system           kube-scheduler-demo-pool1-ec74d5f5-lqkkw               1/1     Running     2          86m
kube-system           rke2-coredns-rke2-coredns-5c84cbd6bd-v9c85             0/1     Unknown     0          85m
kube-system           rke2-coredns-rke2-coredns-autoscaler-78bdc5d6c-b7vrd   0/1     Unknown     0          85m
kube-system           rke2-ingress-nginx-controller-d865t                    1/1     Running     2          80m
kube-system           rke2-metrics-server-5df7d77b5b-4n5vn                   0/1     Unknown     0          81m
tigera-operator       tigera-operator-698786489c-gzvqm                       1/1     Running     2          85m
``` 

Environment:

Harvester ISO version: master-45e25939-head
Rancher version: 2.6.3-rc1
Underlying Infrastructure: one nodes harvester on local vm

**Comment by gitlawr** [comment]
I'm not able to reproduce. Can you provide rke2 logs?

**Comment by guangbochen** [comment]
> I'm not able to reproduce. Can you provide rke2 logs?

This seems to need a 3 nodes cluster, also the issue https://github.com/harvester/harvester/issues/1428 may also seem to be related.

**Comment by TachunLin** [comment]
@gitlawr 
Update the reproduce result on latest harvester master release `master-47ea94ee-head` with rancher `2.6-head`

### Result 
On 3 nodes harvester server machines: 
After fully shutdown, wait more than `15 minutes`, then power on all harvester node machines.
One of my harvester node always failed to launch (`node 3`)
![image](https://user-images.githubusercontent.com/29251855/143991670-81d2dc8e-b61c-4c69-a03c-948af405f816.png)


The other two nodes can start and running correclty

On rancher side, the RKE2 cluster keeps changing between:
`Error:  cluster health check failed`  and `Unavailable: cluster agent is not connected`

![image](https://user-images.githubusercontent.com/29251855/143991982-5c161086-d97a-46f7-a489-eec5e1c0ca51.png)

![image](https://user-images.githubusercontent.com/29251855/143992020-60055f10-e0df-4fb6-9091-1fc52748a116.png)

### Test Environment
* Harvester ISO version: master-47ea94ee-head
* Rancher version: 2.6-head
* Underlying Infrastructure: 3 nodes harvester on local kvm/qemu machine
* Network configuration:
  - VIP: 192.168.122.59 
  - Node1: harvester-n1-i1561 / 192.168.122.142 (Running) 
  - Node2: harvester-n2-i1561 / 192.168.122.158 (Running) 
  - Node3: harvester-n3-i1561 / 192.168.122.172 (Failed) 

### Troubleshooting information
Since node 3 failed to launch, we can't access to harvester. 
Provide relating kubectl output and RKE2 log for further check 

1. RKE2 server log on `failed node 3`
[rke2-server_node3_failed.log](https://github.com/harvester/harvester/files/7623331/rke2-server_node3_failed.log)

2. Check the rke2-coredns-rke2-coredns logs on `running node 2`
![image](https://user-images.githubusercontent.com/29251855/143995256-980018ab-2f58-49b2-9a98-6964f050e74f.png)

3. Check pods status in `harvester-system` namespace on `running node 2`
![image](https://user-images.githubusercontent.com/29251855/143995432-eec412cb-5b97-4c3d-9e78-3c14bed87a98.png)

4. Check pods status in `cattle-system` namespace on  `running node 2`
![image](https://user-images.githubusercontent.com/29251855/143995490-511c071f-d703-44f7-8ff4-c1456f11809e.png)

5. We can access node 3 while can't access node 2 and node 1 through ping command.

### Follow up 
As discussed this morning, I will use one node harvester with RKE2 cluster provisioned to identify the harvester failed to launch behavior and collect more information first.

**Comment by TachunLin** [comment]
Reproduce this issue on `v1.0.0-rc1`, 
After fully shutdown all 3 nodes bare machine and power on after 1 hour. Still encounter the same issue for not being able to launch harvester cluster.  

### Result 
This time can't access harvester dashboard via VIP and unable to ping each node's ip. 


### Status after fully reboot

Node1 console output 
![image](https://user-images.githubusercontent.com/29251855/144851035-a2488fbd-cbbe-47da-88ae-cbf288a81f77.png)

Node2 console output 
![image](https://user-images.githubusercontent.com/29251855/144851117-6cfd75a4-415f-44a0-be40-90ebe42cabe3.png)

Node3 console output 
![image](https://user-images.githubusercontent.com/29251855/144851184-2ce97314-3560-43f5-8609-1f722a47af81.png)

I can get rke2.log by using `journalctl -u rke2-server.service` but no yet able to retrieve it since the harvester node is not reachable, 


Test Environment
* Harvester ISO version: v1.0.0-rc1
* Rancher version: v2.6.0-rc3
* Underlying Infrastructure: 3 nodes harvester on provo bare machines

Network configuration:
* VIP: 10.84.44.181 (Unable to access)
* Node1: 10.84.44.114 (Unable to access)
* Node2: 10.84.44.115 (Unable to access)
* Node3: 10.84.44.116 (Unable to access)

**Comment by TachunLin** [comment]
Try to verify this issue on latest harvester `master-5d6f3db0-head` with rancher `v2.6-head`.
Currently when I try to fully shutdown and power on harvester node machine, I can't access dashboard by VIP.
There is only one node left to be able to access. 

Since harvester did not able to back to work after fully shutdown then power on, thus I can't further validate whether the RKE2 provisioned upon it can recover correctly.   

The detailed steps and available RKE2 log have listed in #1428 for further check and investigation.

**Comment by TachunLin** [comment]
Verified on harvester `master-5d6f3db0-head` without virtual network `vlan 1`

### 
User vlan 806 on provo as a workaround to bypass #1669 

> RKE2 cluster have the ability recover after shutdown all harvester nodes machine for 20 minutes, then power on each nodes in between 25 seconds. 

> RKE2 cluster can also recover when we just reboot one or all of the nodes

![image](https://user-images.githubusercontent.com/29251855/146195487-609c0ede-e7c6-4957-bbf2-57e58a446205.png)

**Please make sure network connection between RKE2 node vm and Rancher if you encounter failed to recovery**
** In case one of your node got failure during startup, once this node can recover with network connectivity, rancher will resume to provision the RKE2 cluster **

![image](https://user-images.githubusercontent.com/29251855/146195303-e07f1cca-06e7-4210-8bbe-8275c02b628b.png)

### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-5d6f3db0-head
* Rancher version: v2.6-head (04520-c0)
* node driver: v0.3.4

### Verify Steps
1. Prepare a 3 nodes harvester cluster (provo bare machine)
2. Enable virtual network with `harvester-mgmt`
3. Create vlan 806 with id `806`
4. Import harvester from rancher and create cloud credential
5. Provision a RKE2 cluster with `vlan 806` 
6. Wait for build up ready
7. Press 'Momentary Press` button to shutdown node 3,
8. Press 'Momentary Press` button to shutdown node 2 
9. Press 'Momentary Press` button to shutdown node 1
10. Wait for 20 minutes
11. Press 'Momentary Press` button to power on node 1, wait 10 seconds
12. Press 'Momentary Press` button to power on node 2, wait 10 seconds
13. Press 'Momentary Press` button to power on node 3
14. Wait for startup complete 

### Support bundle
[supportbundle_29976b55-cbd5-4e7d-a7a2-9c883c9b375c_2021-12-15T12-12-59Z.zip](https://github.com/harvester/harvester/files/7719654/supportbundle_29976b55-cbd5-4e7d-a7a2-9c883c9b375c_2021-12-15T12-12-59Z.zip)


### Additional Context
The verification is based on harvester without vlan 1 since to bypass issue #1669 
We still need to verify the vlan `1` scenario after the issue fixed.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-5e8d9764-head` with rancher `v2.6.3-rc6`. Close this issue.

### Result
After shutdown 3 harvester nodes machine, wait 15 minutes, power on all node machines in sequence, 
RKE2 cluster on harvester `can recover` to `Active` status 

We verified on the following test scenario
1. Fully power cycle 3 nodes harvester on bare-metal machine
2. Fully power cycle 3 nodes harvester on local kvm machine 
3. Reboot all harvester nodes on bare-metal and local kvm machine

> Please be noted, it's based on harvester cluster and rancher connection established without broken 

![image](https://user-images.githubusercontent.com/29251855/146494644-9deb4497-2ff6-4f8f-8b95-35a76d0c9022.png)

1. RKE2 cluster change status after harvester cluster back to work
![image](https://user-images.githubusercontent.com/29251855/146490086-3087cd28-d4b4-4caf-80e9-964a7b114f64.png)

2. Try to rebuild RKE2 node vm
![image](https://user-images.githubusercontent.com/29251855/146490169-3b3b3b56-da02-4336-9fe7-ce940751213f.png)

3. Rebuild RKE2 cluster
![image](https://user-images.githubusercontent.com/29251855/146490196-aea7a846-7f10-43e4-bcbb-659419692a5a.png)



### Test Information
* Test Environment: 3 nodes harvester on provo bare-metal machine and local kvm machines
* Harvester version: master-5e8d9764-head
* Rancher version: v2.6.3-rc6


### Verify Steps

1. Prepare a 3 nodes harvester cluster (provo bare machine)
2. Enable virtual network with harvester-mgmt
3. Create vlan1 with id `1`
4. Import harvester from rancher and create cloud credential
5. Provision a RKE2 cluster with vlan `1`
6. Wait for build up ready
7. Shutdown harvester node 3 
8. Shutdown harvester node 2
9. Shutdown harvester node 1
10. Wait for 20 minutes
11. Power on node 1, wait 10 seconds
12. Power on node 2, wait 10 seconds
13. Power on node 3
14. Wait for harvester startup complete
15. Wait for RKE2 cluster back to work

### Additional Context
If still encounter RKE2 recovery issue, 
You can verify the connectivity from rancher to harvester and vise versa to ensure the connectivity exists

![image](https://user-images.githubusercontent.com/29251855/146490910-3050cc4e-ac9e-4238-aa04-a0a2cb78de7d.png)

```
davidtclin@davidtclin-Standard-PC-Q35-ICH9-2009:~$ sudo docker exec -it e86 /bin/bash
e867598c5144:/var/lib/rancher # curl 192.168.122.158
<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```
